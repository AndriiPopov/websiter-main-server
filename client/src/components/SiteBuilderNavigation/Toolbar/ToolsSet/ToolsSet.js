import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import classes from './ToolsSet.module.css';

export const ToolsSet = props => {
    return (
        <table 
            className={props.activeTopMenuItem === props.title || props.isActive ? [ classes.ToolsSet, classes.active ].join(' ') : classes.ToolsSet}>
            <tbody>
                <tr>
                    {props.children}
                </tr>
            </tbody>
        </table>
    );
};

ToolsSet.propTypes = {
    activeTopMenuItem: PropTypes.oneOf(['sections', 'boxes', 'text'])
};

const mapStateToProps = state => {
    return {
        activeTopMenuItem: state.topMenu.activeTopMenuItem,
    };
};

export default connect(mapStateToProps)(ToolsSet);