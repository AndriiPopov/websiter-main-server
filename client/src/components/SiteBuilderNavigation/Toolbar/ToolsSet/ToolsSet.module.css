.ToolsSet {
    display: none;
    border-collapse: collapse;
    fill: #555;
}

.active {
    display: block;
}