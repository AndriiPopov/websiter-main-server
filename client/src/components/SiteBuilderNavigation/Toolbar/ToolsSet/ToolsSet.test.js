import React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { ToolsSet } from './ToolsSet';

configure({adapter: new Adapter()});

describe('ToolSet', () => {
    it('should be defined', () => {
        expect(ToolsSet).toBeDefined();
    });

    it('it should render a table', () => {
        const tree = shallow(
            <ToolsSet activeTopMenuItem="boxes" />
        );
        expect(toJson(tree)).toMatchSnapshot();
    });
});