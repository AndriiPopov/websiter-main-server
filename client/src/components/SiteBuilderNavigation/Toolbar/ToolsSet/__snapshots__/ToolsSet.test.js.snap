// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolSet it should render a table 1`] = `
<aux>
  <table
    className="ToolsSet"
    key="sections"
  >
    <tbody>
      <tr>
        <Connect(Sections) />
      </tr>
    </tbody>
  </table>
  <table
    className="ToolsSet active"
    key="boxes"
  >
    <tbody>
      <tr>
        <Connect(Boxes) />
      </tr>
    </tbody>
  </table>
</aux>
`;
